### PR TITLE
Add accounts

### PR DESCRIPTION
**Pull request by cl3t0**
### 💡 Summary

This PR aims to create accounts and related tables and make necessary changes in the database.

### 📝 Details

- [x] Create queries and mutations in the GraphQL layer;
- [x] Create entities for accounts;
- [x] Implement necessary services;
- [x] Add the account table and some auxiliary tables;
- [x] Implement necessary model;

### 🔎 References

base branch: develop

### CODE REVIEW

File: src/models/account.rs [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+    diesel::insert_into(account_schema::table)
+        .values(&parsed_account)
+        .get_result::<Account>(&mut conn.get()?)
+        .map(|t| t.to_entity())
+        .map_err(AccountModelError::FailedToCreateAccount)
+}
+
+pub fn delete_account(conn: &database::DbPool, id: &Uuid) -> Result<()> {
+    diesel::delete(account_schema::table.filter(account_schema::id.eq(id)))
+        .execute(&mut conn.get()?)
```
Comment (davisuga) [inline src/models/account.rs]: Please soft delete the account record

File: src/utils/mod.rs [hunk 2]
```diff
@@ -4,3 +4,7 @@ pub fn first_or<T>(a: Option<T>, b: Option<T>) -> Option<T> {
         None => b,
     }
 }
+
+pub fn invert<T, E>(x: Option<Result<T, E>>) -> Result<Option<T>, E> {
+    x.map_or(Ok(None), |v| v.map(Some))
+}
```
Comment (davisuga) [inline src/utils/mod.rs]: Function naming suggestion:
```suggestion
pub fn opt_result_of_result_opt<T, E>(x: Option<Result<T, E>>) -> Result<Option<T>, E> {
    x.map_or(Ok(None), |v| v.map(Some))
}
```

**Review by davisuga — CHANGES_REQUESTED** [review]
Almost perfect, good job!

File: src/models/account.rs (hunk 1 above)
Comment (cl3t0) [reply src/models/account.rs]: Done

File: src/utils/mod.rs (hunk 2 above)
Comment (cl3t0) [reply src/utils/mod.rs]: Done

**Review by davisuga — APPROVED** [review]
LGTM!

**Pull request merged**